File: PhoneBook.java
import java.util.ArrayList;
import java.util.Scanner;

public class PhoneBook {
    public static ArrayList<String> phoneBook = new ArrayList<>();
    public static Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        while (true) {
            printDisplayMenu();
            System.out.println("WELCOME TO JOHNSON PHONEBOOK RECORD");
            System.out.println("Choose an option");
            int option = scanner.nextInt();
            scanner.nextLine();
            switch (option) {
                case 1:
                    addContact();
                    break;
                case 2:
                    removeContact();
                    break;
                case 3:
                    findContactByPhoneNumber();
                    break;
                case 4:
                    findContactByFirstName();
                    break;
                case 5:
                    findContactByLastName();
                    break;
                case 6:
                    editContact();
                    break;
                case 7:
                    System.exit(0);
                    break;
                default:
                    System.out.println("Invalid choice!");
            }
        }
    }

    public static void addContact() {
        System.out.println("Enter first name");
        String firstName = scanner.nextLine();
        System.out.println("Enter last name");
        String lastName = scanner.nextLine();
        System.out.println("Enter phone number");
        String phoneNumber = scanner.nextLine();
        phoneBook.add(firstName + " " + lastName + " " + phoneNumber);
        System.out.println("Contact added successfully!");
    }

    public static void removeContact() {
        System.out.println("Enter first name");
        String firstName = scanner.nextLine();
        System.out.println("Enter last name");
        String lastName = scanner.nextLine();
        for (String contact : phoneBook) {
            if (contact.contains(firstName + " " + lastName)) {
                phoneBook.remove(contact);
                System.out.println("Contact removed successfully!");
                return;
            }
        }
        System.out.println("Contact not found!");
    }

    public static void findContactByPhoneNumber() {
        System.out.println("Enter phone number");
        String phoneNumber = scanner.nextLine();
        for (String contact : phoneBook) {
            if (contact.contains(phoneNumber)) {
                System.out.println(contact);
                return;
            }
        }
        System.out.println("Contact not found!");
    }

    public static void findContactByFirstName() {
        System.out.println("Enter first name");
        String firstName = scanner.nextLine();
        for (String contact : phoneBook) {
            if (contact.startsWith(firstName)) {
                System.out.println(contact);
                return;
            }
        }
        System.out.println("Contact not found!");
    }

    public static void findContactByLastName() {
        System.out.println("Enter last name");
        String lastName = scanner.nextLine();
        for (String contact : phoneBook) {
            if (contact.contains(" " + lastName)) {
                System.out.println(contact);
                return;
            }
        }
        System.out.println("Contact not found!");
    }

    public static void editContact() {
        System.out.println("Enter first name");
        String firstName = scanner.nextLine();
        System.out.println("Enter last name");
        String lastName = scanner.nextLine();
        for (String contact : phoneBook) {
            if (contact.contains(firstName + " " + lastName)) {
                System.out.println("Enter new first name");
                String newFirstName = scanner.nextLine();
                System.out.println("Enter new last name");
                String newLastName = scanner.nextLine();
                System.out.println("Enter new phone number");
                String newPhoneNumber = scanner.nextLine();
                phoneBook.remove(contact);
                phoneBook.add(newFirstName + " " + newLastName + " " + newPhoneNumber);
                System.out.println("Contact updated successfully!");
                return;
            }
        }
        System.out.println("Contact not found!");
    }

    public static void printDisplayMenu() {
        System.out.println("1. Add Contact");
        System.out.println("2. Remove Contact");
        System.out.println("3. Find Contact by Phone Number");
        System.out.println("4. Find Contact by First Name");
        System.out.println("5. Find Contact by Last Name");
        System.out.println("6. Edit Contact");
        System.out.println("7. Exit");
    }
}

File: phonebook.py
phone_book = []
def print_display_menu():
    print("1. Add Contact")
    print("2. Remove Contact")
    print("3. Find Contact by Phone Number")
    print("4. Find Contact by First Name")
    print("5. Find Contact by Last Name")
    print("6. Edit Contact")
    print("7. Exit")

def add_contact():
    first_name = input("Enter first name: ")
    last_name = input("Enter last name: ")
    phone_number = input("Enter phone number: ")
    phone_book.append(f"{first_name} {last_name} {phone_number}")
    print("Contact added successfully!")

def remove_contact():
    first_name = input("Enter first name: ")
    last_name = input("Enter last name: ")
    for contact in phone_book:
        if first_name + " " + last_name in contact:
            phone_book.remove(contact)
            print("Contact removed successfully!")
            return 
    print("Contact not found!")

def find_contact_by_phone_number():
    phone_number = input("Enter phone number: ")
    for contact in phone_book:
        if phone_number in contact:
            print(contact)
            return
    print("Contact not found!")

def find_contact_by_first_name():
    first_name = input("Enter first name: ")
    for contact in phone_book:
        if contact.startswith(first_name):
            print(contact)
            return
    print("Contact not found!")

def find_contact_by_last_name():
    last_name = input("Enter last name: ")
    for contact in phone_book:
        if " " + last_name in contact:
            print(contact)
            return
    print("Contact not found!")

def edit_contact():
    first_name = input("Enter first name: ")
    last_name = input("Enter last name: ")
    for contact in phone_book:
        if first_name + " " + last_name in contact:
            new_first_name = input("Enter new first name: ")
            new_last_name = input("Enter new last name: ")
            new_phone_number = input("Enter new phone number: ")
            phone_book.remove(contact)
            phone_book.append(f"{new_first_name} {new_last_name} {new_phone_number}")
            print("Contact updated successfully!")
            return
    print("Contact not found!")

while True:
    print_display_menu()
    print("WELCOME TO JOHNSON PHONEBOOK RECORD")
    print("Choose an option")
    option = int(input())
    if option == 1:
        add_contact()
    elif option == 2:
        remove_contact()
    elif option == 3:
        find_contact_by_phone_number()
    elif option == 4:
        find_contact_by_first_name()
    elif option == 5:
        find_contact_by_last_name()
    elif option == 6:
        edit_contact()
    elif option == 7:
        break
    else:
        print("Invalid choice!")

